tools/fog-local-network/fog_conformance_tests.py

```diff
@@ -869,35 +1003,95 @@ def run(self, args):
         # Take 2 from wallet0
-        credits9 = [
+        credits15 = [
             {'account': 0, 'amount': 1},
             {'account': 1, 'amount': 1},
             {'account': 2, 'amount': 1},
             {'account': 3, 'amount': 1},
             {'account': 4, 'amount': 1},
         ]
-        key_images9 = [block8_key_images[0]]
-        print("Key images spent in Block 9: ", key_images9)
-        block9_key_images = ledger1.add_block(credits9, key_images9, fog_pubkey)
-        ledger2.add_block(credits9, key_images9, fog_pubkey)
-        print("Key images for new transactions in Block 9: ", block9_key_images)
+        key_images15 = [block14_key_images[0]]
+        print("Key images spent in Block 15: ", key_images15)
+        block15_key_images = ledger1.add_block(credits15, key_images15, fog_pubkey)
+        ledger2.add_block(credits15, key_images15, fog_pubkey)
+        print("Key images for new transactions in Block 15: ", block15_key_images)
         time.sleep(1)
 
         # Check balances. These should come from the new RNG of the second ingest server
-        self.multi_balance_checker.balance_check("from9", [
-            [{9: 15, 10: 14}, [10]],
-            [{9: 12, 10: 13}, [10]],
-            [{9: 3, 10: 4}, [10]],
-            [{9: 3, 10: 4}, [10]],
-            [{9: 12, 10: 13}, [10]],
+        self.multi_balance_checker.balance_check("from15", [
+            [{15: 15, 16: 14}, [16]],
+            [{15: 12, 16: 13}, [16]],
+            [{15: 3, 16: 4}, [16]],
+            [{15: 3, 16: 4}, [16]],
+            [{15: 16, 16: 17}, [16]],
+        ])
+
+        # We told the server to retire at block 13, and b/c the current pubkey expiry is 3,
+        # we need to get to block 16 in order to get the ingest server to retire.
+
+        # Add block 16. This should cause ingest1 to become Idle.
+        # Give 1 to everyone
+        # Take 1 from 0
+        credits16 = [
+            {'account': 0, 'amount': 1},
+            {'account': 1, 'amount': 1},
+            {'account': 2, 'amount': 1},
+            {'account': 3, 'amount': 1},
+            {'account': 4, 'amount': 1},
+        ]
+        key_images16 = [block15_key_images[0]]
+        print("Key images spent in Block 16: ", key_images16)
+        block16_key_images = ledger1.add_block(credits16, key_images16, fog_pubkey)
+        ledger2.add_block(credits16, key_images16, fog_pubkey)
+        print("Key images for new transactions in Block 16: ", block16_key_images)
+        time.sleep(3)
+
+        self.multi_balance_checker.balance_check("from16", [
+            [{16: 14, 17: 14}, [17]],
+            [{16: 13, 17: 14}, [17]],
+            [{16: 4, 17: 5}, [17]],
+            [{16: 4, 17: 5}, [17]],
+            [{16: 17, 17: 18}, [17]],
         ])
 
+        # TODO: Expecting the server to be Idle here, but it's still Active...
+        status = self.fog_ingest.get_status()
+        print("Ingest server status after block 16: ", status)
```

Comment:
This is an outstanding issue....After block 16 is published, I was expecting the server to retire and be Idle here, but the status still reports it as Active.  Not sure why or what's missing.